core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala

```diff
@@ -61,7 +61,8 @@ class TimeSeriesPartition(val partID: Int,
                           // Shared class for mutating the infoMap / OffheapLFSortedIDMap given mapPtr above
                           offheapInfoMap: OffheapLFSortedIDMapMutator,
                           // Lock state used by OffheapLFSortedIDMap.
-                          var lockState: Int = 0)
+                          var lockState: Int = 0,
+                          var lastTime: Long = -1L)
```

Comment:
I assume you are not fetching the lastTime from writeBuffers or chunks due to performance reasons? Is that confirmed with a benchmark? Perhaps we can document it here.

Reply:
lastChunkInfo.lastTime should provide us this information. We shouldnt need to unwrap/decompress the chunk to get the lastTime.

Reply:
lastChunkInfo is potentially expensive is the problem.  Locks are involved and it has to consult the chunkMap.  Storing the lastTime, yes it uses an extra long but is much simpler and faster.  I can document this.